ci/ray_ci/doc/autodoc.py

```diff
@@ -55,11 +56,28 @@ def _get_autodoc_rsts(self) -> List[str]:
         if self._autodoc_rsts is not None:
             return self._autodoc_rsts
 
-        dir = os.path.dirname(self._head_rst_file)
-        self._autodoc_rsts = [self._head_rst_file]
-        with open(self._head_rst_file, "r") as f:
+        self._autodoc_rsts = {self._head_rst_file}
+        visited = set()
+        while len(self._autodoc_rsts) > len(visited):
+            not_visited = self._autodoc_rsts - visited
+            for rst in not_visited:
+                visited.add(rst)
+                self._autodoc_rsts.update(self._get_autodoc_rsts_from_single_file(rst))
+
+        return self._autodoc_rsts
+
+    def _get_autodoc_rsts_from_single_file(self, rst_file: str) -> Set[str]:
+        rsts = set()
+        dir = os.path.dirname(rst_file)
+        with open(rst_file, "r") as f:
             line = f.readline()
             while line:
+                # look for the include block
+                if line.strip().startswith(_SPHINX_INCLUDE_HEADER):
```

Comment:
just add `line = line.strip()` before this? so that you do not need to `strip` it at least twice in all conditions?

and if you stripped it, then the next line is just removing the `prefix`, which can use a substring / slicing, rather than split with `[1]` ? otherwise, there might be cases where if the line contains multiple `::` (maybe in an error case or something?), and it will lead to weird results?